rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -269,15 +232,14 @@ class Publisher : public PublisherBase
               "intra process publish called after destruction of intra process manager");
     }
     if (!msg) {
-      throw std::runtime_error("cannot publisher msg which is a null pointer");
+      throw std::runtime_error("cannot publish msg which is a null pointer");
     }
-    uint64_t message_seq =
-      ipm->template store_intra_process_message<MessageT, Alloc>(publisher_id, msg);
-    return message_seq;
+
+    ipm->template do_intra_process_publish<MessageT>(publisher_id, std::move(msg));
   }
 
-  uint64_t
-  store_intra_process_message(
+  void
+  do_intra_process_publish(
     uint64_t publisher_id,
```

Comment:
The `publisher_id` is a member variable of the class, the argument in the function isn't needed.
I would just use `intra_process_publisher_id_`.

Reply:
Actually that's my error, because you haven't changed the signature. If you change this, also do it in the other overload above.

Reply:
Done